logbook-core/src/main/java/org/zalando/logbook/HeaderFilters.java

```diff
@@ -30,13 +29,24 @@ public static HeaderFilter replaceHeaders(final BiPredicate<String, String> pred
         return eachHeader((key, value) -> predicate.test(key, value) ? replacement : value);
     }
 
+    public static HeaderFilter removeHeaders(final Predicate<String> keyPredicate) {
+        return eachHeader((key, value) -> keyPredicate.test(key) ? null : value);
+    }
+
+    public static HeaderFilter removeHeaders(final BiPredicate<String, String> predicate) {
+        return eachHeader((key, value) -> predicate.test(key, value) ? null : value);
+    }
+
     public static HeaderFilter eachHeader(final BinaryOperator<String> operator) {
         return headers -> {
             final BaseHttpMessage.HeadersBuilder result = new BaseHttpMessage.HeadersBuilder();
 
             for (final Map.Entry<String, List<String>> e : headers.entrySet()) {
                 final String k = e.getKey();
-                result.put(k, e.getValue().stream().map(x -> operator.apply(k, x)).collect(toList()));
```

Comment:
I initially implemented it as a a pair of modify-operator and remove-predicate, but then I realized that it is always either modification or removal so I went with considering nulls as removal sign. I would say that null contract was no well documented, that is why it is not strictly a breaking change.